app/livechat/server/api/v1/videoCall.js

```diff
@@ -56,3 +59,102 @@ API.v1.addRoute('livechat/video.call/:token', {
 		}
 	},
 });
+
+API.v1.addRoute('livechat/webrtc.call', { authRequired: true }, {
+	get() {
+		try {
+			check(this.queryParams, {
+				rid: Match.Maybe(String),
+			});
+
+			if (!hasPermission(this.userId, 'view-l-room')) {
+				return API.v1.unauthorized();
+			}
+
+			const room = canSendMessage(this.queryParams.rid, {
+				uid: this.userId,
+				username: this.user.username,
+				type: this.user.type,
+			});
+			if (!room) {
+				throw new Meteor.Error('invalid-room');
+			}
+
+			const webrtcCallingAllowed = (rcSettings.get('WebRTC_Enabled') === true) && (rcSettings.get('Omnichannel_call_provider') === 'WebRTC');
+			if (!webrtcCallingAllowed) {
+				throw new Meteor.Error('webRTC calling not enabled');
+			}
+
+			const config = settings();
+			if (!config.theme || !config.theme.actionLinks || !config.theme.actionLinks.webrtc) {
+				throw new Meteor.Error('invalid-livechat-config');
+			}
+
+			if (!room.callStatus || room.callStatus === 'ended' || room.callStatus === 'declined') {
+				Rooms.setCallStatus(room._id, 'ringing');
+				Messages.createWithTypeRoomIdMessageAndUser(
+					'livechat_webrtc_video_call',
+					room._id,
+					'Join my room to start the video call',
+					this.user,
+					{
+						actionLinks: config.theme.actionLinks.webrtc,
+						callStatus: 'ringing',
+
+					},
+				);
+			}
+			const videoCall = {
+				rid: room._id,
+				provider: 'webrtc',
+				callStatus: room.callStatus,
```

Comment:
I don't think the `room` object here has the current data. Incase the following condition is TRUE - `!room.callStatus || room.callStatus === 'ended' || room.callStatus === 'declined'` - you're updating the callStatus to `ringing` but the original object still contains stale data